File: Linear Data Structures/assignment/Stack.py
class Node:
    def __init__(self, val=None, next=None):
        self.val = val
        self.next = next

class Stack:
    def __init__(self, head=None):
        self.head = head

    # Push a value into the stack
    def push(self, n):
        if self.head is None:
            self.head = Node(n)
            return

        newNode = Node(n)
        newNode.next = self.head
        self.head = newNode

    # Pop the latest inserted value from the stack
    # If stack is empty, return any message
    def pop(self):
        if self.head is None:
            return "Can't pop. Stack is empty."

        temp = self.head
        del self.head
        self.head = temp.next

        return temp.val

    # return the latest value of the stack
    def peek(self):
        return self.head.val

    # Check if stack is empty
    # return True if stack is empty, else False
    def isEmpty(self):
        if self.head is not None:
            return False
        
        return True

    # Prints all of the value(s) in the stack
    def __str__(self):
        curr = self.head
        s = ""

        while curr:
            s += str(curr.val) + " "
            curr = curr.next

        return s

File: Linear Data Structures/solutions/Java/Stack.java
/*
 * This is a Stack Data Structure implementation using LinkedList
 */
public class Stack {

    static class Node {
        Object value;
        Node next;

        Node(Object value) {
            this.value = value;
        }
    }

    private Node head;

    // Constructor
    public Stack() {
        this.head = null;
    }

    /*
     * Insert an element into the stack
     */
    public void push(Object n) {
        Node newNode = new Node(n);

        if (this.head == null) {
            this.head = newNode;
        } else {
            newNode.next = this.head;
            this.head = newNode;
        }
    }

    /*
     * Remove an element into the stack (LIFO)
     */
    public void pop() {
        if (this.head == null) {
            System.out.println("Cannot pop an element from stack. Stack is empty.");
        } else {            
            this.head = this.head.next;
        }
    }

    /*
     * @return the latest element inside the stack
     */
    public Object peek() {
        return this.head.value;
    }

    /*
     * Check if the stack is empty
     * @return true if it is empty, else false
     */
    public boolean isEmpty() {
        if (this.head == null) {
            return true;
        }

        return false;
    }

    @Override
    /*
     * Prints all of the value inside stack
     */
    public String toString() {
        Node curr = this.head;
        String s = "";

        while (curr != null) {
            s += curr.value + " ";
            curr = curr.next;
        }

        return s;
    }
}
